pkgs/tools/virtualization/nixos-nspawn/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, lib, python3, nixosTests }:
+
+stdenv.mkDerivation rec {
+  name = "nixos-nspawn";
+  src = lib.cleanSourceWith {
+    src = ./.;
+    filter = path: type:
+      type == "regular" && ! (lib.hasSuffix "default.nix" path);
+  };
+
+  buildInputs = [ python3 python3.pkgs.wrapPython ];
+
+  postPatch = ''
+    sed -i -e '1,2d' ./nixos-nspawn.py
+    sed -i -e '1s;^;#!${python3}/bin/python3;' ./nixos-nspawn.py
+
+    substituteInPlace ./nixos-nspawn.py \
+      --replace "@eval@" "${./eval-container.nix}"
+  '';
+
+  dontBuild = true;
+
+  passthru.tests = { inherit (nixosTests) containers-next-imperative; };
+
+  doCheck = true;
+  checkInputs = [ python3.pkgs.flake8 ];
+  checkPhase = ''
+    flake8 ./nixos-nspawn.py --max-line-length 130 \
+      --ignore E722,W503
+  '';
```

Comment:
```suggestion
```
flake8 shouldn't be run in checkPhase like mypy because it tends to break the packages when flake8 is updated and they change some existing check or added a new one which happens from time to time even in minor releases.